build/ci/pr/Jenkinsfile

```diff
@@ -90,3 +104,16 @@ def notOnlyDocs() {
     	returnStatus: true
     ) == 0
 }
+
+def ifImageNeedsRebuilding() {
```

Comment:
It seems to me we could `docker pull || docker build && docker push` in the `ci-build-image` target and remove this `ifImageNeedsRebuilding` func?